Tests/SuperLinq.Test/FallbackIfEmptyTest.cs

```diff
@@ -7,14 +7,58 @@ public void FallbackIfEmptyWithEmptySequence()
 	{
 		using var source = Enumerable.Empty<int>().AsTestingSequence(maxEnumerations: 2);
 		source.FallbackIfEmpty(12).AssertSequenceEqual(12);
-		source.FallbackIfEmpty(12, 23).AssertSequenceEqual(12, 23);
+	}
+
+	[Fact]
+	public void FallbackIfEmptyWithCollectionSequence()
+	{
+		using var source = Enumerable.Empty<int>().AsTestingCollection(maxEnumerations: 2);
+		source.FallbackIfEmpty(12).AssertSequenceEqual(12);
 	}
 
 	[Fact]
 	public void FallbackIfEmptyWithNotEmptySequence()
 	{
 		using var source = Seq(1).AsTestingSequence(maxEnumerations: 2);
-		source.FallbackIfEmpty(12).AssertSequenceEqual(1);
-		source.FallbackIfEmpty(12, 23).AssertSequenceEqual(1);
+		source.FallbackIfEmpty(new BreakingSequence<int>()).AssertSequenceEqual(1);
+	}
+
+	[Fact]
+	public void FallbackIfEmptyWithNotEmptyCollectionSequence()
+	{
+		using var source = Seq(1).AsTestingCollection(maxEnumerations: 2);
+		source.FallbackIfEmpty(new BreakingSequence<int>()).AssertSequenceEqual(1);
+	}
+
+	[Fact]
+	public void AssertFallbackIfEmptyCollectionBehaviorOnEmptyCollection()
+	{
+		using var source = Enumerable.Empty<int>().AsTestingCollection(maxEnumerations: 2);
+
+		var result = source.FallbackIfEmpty(12);
+		result.AssertCollectionErrorChecking(1);
+	}
+
+	[Fact]
+	public void AssertFallbackIfEmptyCollectionBehaviorOnNonEmptyCollection()
+	{
+		using var source = Seq(1).AsTestingCollection(maxEnumerations: 2);
+
+		var result = source.FallbackIfEmpty(new BreakingSequence<int>());
+		result.AssertCollectionErrorChecking(1);
+	}
+
+	[Fact]
+	public void FallbackIfEmptyUsesCollectionCountAtIterationTime()
+	{
+		using var source = new BreakingCountCollection<int>();
```

Comment:
We don't need to use `BreakingCountCollection` here. Simply follow the example in other `CollectionCountAtIterationTime` methods. Create a stack and set up the result with a sequence. Test that the result sequence is the fallback sequence. Then add a value to the stack, and evaluate result again that it uses the stack sequence instead of the fallback sequence.

The corner case we are testing here is that the behavior of `FallbackIfEmpty` is not determined by the value of `Count` when `FallbackIfEmpty` is called, but when it is enumerated. See morelinq/morelinq#945 for explanation of error sequence. I see that we are evaluating correctly here, but we should validate the future changes don't break this behavior.